init_and_run_app.sh

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/env bash
+
+# NAME: init_and_run_app.sh
+#
+# BRIEF: This script installs packages needed in the environment to run the
+# See Stuff application and then runs it locally.
+
+check_current_directory () {
+  if [[ ! -d '../rails_see_stuff' ]]; then
+    echo 'You must be in the directory, rails_see_stuff, after cloning the code'
+    echo 'from https://github.com/kimlew/rails_see_stuff'
+    exit 1
+  fi
+}
+
+check_if_rbenv_installed () {
+  if ! command -v rbenv > /dev/null; then
+    echo 'You must install the package manager, rbenv. For example, install with:'
+    echo 'sudo apt install rbenv OR'
+    echo 'brew install rbenv ruby-build'
+    exit 1
+  fi
+}
+
+check_current_directory
+check_if_rbenv_installed
+
+# Use specific version of Ruby for project that works with Rails 7.
+# Install bundler which is required for rake commands for the database & items.
+rbenv local 3.1.2
+gem install bundler -v 2.3.22
```

Comment:
I had a problem with this line after I installed `rbenv` and used it to install Ruby 3.1.2, because I had not run `eval $(rbenv init -)` anywhere. Because of this, `gem` resolved to my system Ruby's `gem`, not the one in the `rbenv`-installed Ruby 3.1.2, and the script failed because it couldn't install the `bundler` gem. That's _good_ actually, but it might trip people up.

The simplest way to make sure that `gem` is the one from the `rbenv`-installed Ruby is to run `rbenv init` in this script like this, before you try running `gem`:

```
eval $(rbenv init - bash)
```

Reply:
Thinking this over more, you should also put the process expansion in double quotes:

```
eval "$(rbenv init - bash)"
```

because the result of `rbenv init - bash` will definitely contain whitespace.

Reply:
If you want to see what `eval` will evaluate, run `rbenv init - bash` at your terminal window. You will see that it prints out some shell code... one line sets and exports `PATH` and so on. Most of it is code to define an `rbenv` shell function.

Reply:
Added. Thanks.

Reply:
Double quotes added around process expansion.

I tried out:
mac$ eval "$(rbenv init - bash)"
mac$ echo $?
0

But it is good to see what actually runs too.